src/check_zfs.py

```diff
@@ -0,0 +1,126 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import argparse
+from subprocess import check_output, STDOUT, CalledProcessError
+import platform
+import re
+import sys
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=('Check raid state, fragmentation and utilisation of ZPools'),
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        '--fragmentation',
+        help="thresholds for fragmentation of zpool",
+        metavar=('warning', 'critical'),
+        type=int, nargs=2, default=[30, 50],
+    )
+
+    parser.add_argument(
+        '--capacity',
+        help="thresholds for space utilization of zpool",
+        metavar=('warning', 'critical'),
+        type=int, nargs=2, default=[70, 90],
+    )
+
+    args = parser.parse_args()
+
+    try:
+        proc = check_output(
+            ['/sbin/zpool', 'list', '-H'], stderr=STDOUT
+        ).decode()
+    except OSError as e:
+        print('UNKNOWN: can\'t read zpool status: {}'.format(e))
+        return ExitCodes.unknown
+    except CalledProcessError as e:
+        print('UNKNOWN: can\'t read zpool status: {}'.format(e.output))
+        return ExitCodes.unknown
+    
+    exit_multiline = []
+    lines = proc.splitlines()
+
+    if not lines:
+        print('No zpools found!')
+        return(ExitCodes.unknown)
+
+    exit_code = ExitCodes.ok
+    exit_header = 'All zpools are fine'
+    for line in lines:
+        zpool_code, zpool_name, zpool_state = parse_zpool(line, args)
+        exit_multiline.append('{}: {}'.format(zpool_name, zpool_state))
+        if zpool_code > exit_code:
+            exit_code = zpool_code
+            exit_header = 'Some zpools have issues!'
+
+    print(exit_header)
+    if exit_multiline:
+        print('\n'.join(exit_multiline))
+    return exit_code
+
+
+def parse_zpool(line, args):
+    """
+    The output looks like this, with -H there are real tabs separating fields:
+
+    # sudo zpool list -H
+    tank\t130T\t75.9T\t54.1T\t-\t22%\t58%\t1.00x\tONLINE\t-
+    ...
+    """
+
+    name, _, _, _, _, frag, cap, _, health, _ = line.split('\t')
+
+    ret_code = ExitCodes.ok
+    cap = int(cap.split('%')[0])
+    frag = int(frag.split('%')[0])
+
+    if (
+        health != 'ONLINE' or
+        frag > args.fragmentation[0] or
+        cap > args.capacity[0]
+    ):
+        ret_code = ExitCodes.warning
+
+    if (
+        frag > args.fragmentation[1] or
+        cap > args.capacity[1]
+    ):
+        ret_code = ExitCodes.critical
+
+    return (ret_code, name, "health: {}, fragmentation: {}%, capacity: {}%".format(health, frag, cap))
```

Comment:
By separating this it's easier to read and understand for me.
```suggestion

    message = 'health: {}, fragmentation: {}%, capacity: {}%'
    return (ret_code, name, message.format(health, frag, cap))
```

Reply:
Fixed.